### PR TITLE
Issue #3648 - SimpleContainerScope should use WebSocket Behavior for SSL defaults

### PR DESCRIPTION
**Pull request by joakime**
Signed-off-by: Joakim Erdfelt <joakim.erdfelt@gmail.com>

### CODE REVIEW

File: jetty-websocket/websocket-common/src/main/java/org/eclipse/jetty/websocket/common/scopes/SimpleContainerScope.java [hunk 1]
```diff
@@ -166,4 +133,44 @@ public void removeSessionListener(WebSocketSessionListener listener)
...
+            {
+                this.sslContextFactory = new SslContextFactory.Server();
+            }
+        }
+
+        if (executor == null)
+        {
+            QueuedThreadPool threadPool = new QueuedThreadPool();
+            String behavior = "Container";
+            if (policy != null)
```
Comment (sbordet) [inline jetty-websocket/websocket-common/src/main/java/org/eclipse/jetty/websocket/common/scopes/SimpleContainerScope.java]: Why this null check here? You would have already gotten a NPE at line 142 above, no?

File: jetty-websocket/websocket-common/src/main/java/org/eclipse/jetty/websocket/common/scopes/SimpleContainerScope.java [hunk 2]
```diff
@@ -166,4 +133,44 @@ public void removeSessionListener(WebSocketSessionListener listener)
     {
         return sessionListeners;
     }
+
+    @Override
+    protected void doStart() throws Exception
```
Comment (sbordet) [inline jetty-websocket/websocket-common/src/main/java/org/eclipse/jetty/websocket/common/scopes/SimpleContainerScope.java]: I don't think it's right to move the initialization code that before was in the constructor to `doStart()`.
For example, `WebSocketClient` constructor calls `HttpClientProvider.get(scope)` passing in a scope that is not yet started, which get passed to `DefaultHttpClientProvider.newHttpClient(scope)` which uses the scope before it's started.
I think we need a better review of the lifecycle of the scope to move things to a different place.

**Comment by joakime** [comment]
`SimpleContainerScope` refactored heavily.

+ client specific behaviors moved to `ClientContainerScope` (no longer a lifecycle)
+ the rest of the `SimpleContainerScope` has been moved to `src/test/java` (also no longer a lifecycle)
+ the interface `WebSocketContainerScope` has been simplified a bit.

**Comment by joakime** [comment]
I am going to submit 3 separate PRs for this.

1. The simple fix for the limited scope of the issue #3648.
2. The more comprehensive fix for scopes / containers via issue #3730
3. Test cases for the 3 different ways CDI can operate via issue #3731

**Pull request closed without merging**